aiida_quantumespresso/common/workchain/base/restart.py

```diff
@@ -215,71 +200,43 @@ def on_terminated(self):
+        adapt the inputs as an error handler would and return `False`. This will signal to the work chain that a new
+        calculation should be started. If `None` is returned, the work chain assumes that the outputs produced by the
+        calculation are good and nothing will be done.
+
+        :param calculation: the calculation whose outputs should be checked for consistency
+        :return: `ExitCode` if the work chain is to be aborted, `False` if a new calculation should be launched
         """
-        Calculations that run successfully may still have problems that can only be determined when inspecting
-        the output. The same problems may also be the hidden root of a calculation failure. For that reason,
-        after verifying that the calculation ran, regardless of its calculation state, we perform some sanity
-        checks.
-        """
-        pass
-
-    def _handle_submission_failure(self, calculation):
-        """
-        The submission of the calculation has failed. If the submission_failure flag is set to true, this
-        is the second consecutive submission failure and we abort the workchain Otherwise we restart once more.
-        """
-        if self.ctx.submission_failure:
-            self.report('submission for {}<{}> failed for the second consecutive time'
-                .format(self.ctx.calc_name, calculation.pk))
-            return self.exit_codes.ERROR_SECOND_CONSECUTIVE_SUBMISSION_FAILURE
-
-        else:
-            self.report('submission for {}<{}> failed, restarting once more'
-                .format(self.ctx.calc_name, calculation.pk))
 
-    def _handle_unexpected_failure(self, calculation, exception=None):
-        """
-        The calculation has failed for an unknown reason and could not be handled. If the unexpected_failure
-        flag is true, this is the second consecutive unexpected failure and we abort the workchain.
-        Otherwise we restart once more.
-        """
-        if exception:
-            self.report('{}'.format(exception))
-
-        if self.ctx.unexpected_failure:
-            self.report('failure of {}<{}> could not be handled for the second consecutive time'
-                .format(self.ctx.calc_name, calculation.pk))
-            return self.exit_codes.ERROR_SECOND_CONSECUTIVE_UNHANDLED_FAILURE
+    def _handle_calculation_failure(self, calculation):
+        """Call the attached error handlers if any to attempt to correct the cause of the calculation failure.
 
-        else:
-            self.report('failure of {}<{}> could not be handled, restarting once more'
-                .format(self.ctx.calc_name, calculation.pk))
+        The registered error handlers will be called in order based on their priority until a handler returns a report
+        that instructs to break. If the last executed error handler defines an exit code, that will be returned to
+        instruct the work chain abort. Otherwise the work chain will continue the cycle.
```

Comment:
'to instruct the work chain **to** abort'

Reply:
Thanks, will correct it